spark/v3.3/build.gradle

```diff
@@ -258,7 +261,7 @@ project(":iceberg-spark:iceberg-spark-runtime-${sparkMajorVersion}_${scalaVersio
     relocate 'io.airlift', 'org.apache.iceberg.shaded.io.airlift'
     relocate 'org.apache.httpcomponents.client5', 'org.apache.iceberg.shaded.org.apache.httpcomponents.client5'
     // relocate Arrow and related deps to shade Iceberg specific version
-    relocate 'io.netty.buffer', 'org.apache.iceberg.shaded.io.netty.buffer'
+    relocate 'io.netty', 'org.apache.iceberg.shaded.io.netty'
```

Comment:
I think we should be specific about our netty modules we are relocating here. We talked offline and ideally I wish we had a arrow distribution which shaded netty so we wouldn't have to play this game of excluding netty everywhere then adding back in a specific runtime version just to get the shading correctly. 

That said for now I think just being explicit will help us in the future to remember what we are doing here 🤞

Reply:
That's fair. Again I wish we had an arrow distro that just did this for us, but such is life.

Reply:
Ugh, it is really unfortunate.

If I relocate everything under `io.netty`, it won't be safe if we start pulling netty libs from somewhere. If I relocate `io.netty.util`, nobody can guarantee a future version of nitty-common won't have classes in another package.

Let's probably keep the wildcard relocate as at least we are in control of our deps.